### PR TITLE
Kill firemodes

### PR DESCRIPTION
**Pull request by ObaniGemini**
"FIRE_MODES_COUNT" is still there in `gs_public.h` but I don't know what it refers to.

Didn't remove WEAP_ANIM_STRONG and WEAP_ANIM_WEAK because I don't want to mess with anims for now

### CODE REVIEW

File: source/game/g_utils.cpp [hunk 1]
```diff
@@ -1872,8 +1868,7 @@ static bool G_LoadFiredefFromFile( int weapon, firedef_t *firedef ) {
 		return false;
 	}
 
-	Q_snprintfz( filename, sizeof( filename ), "weapondefs/%s %s.def", GS_FindItemByTag( weapon )->shortname,
-				 ( firedef->fire_mode == FIRE_MODE_STRONG ) ? "strong" : "weak" );
+	Q_snprintfz( filename, sizeof( filename ), "weapondefs/%s .def", GS_FindItemByTag( weapon )->shortname );
```
Comment (mikejsavage) [inline source/game/g_utils.cpp]: tbh delete this whole WEAPONDEFS_FROM_DISK thing

File: source/gameshared/gs_public.h [hunk 2]
```diff
@@ -660,13 +660,6 @@ enum {
 	PAIN_TOTAL
 };
 
-//fire modes as event parm
-enum {
-	FIRE_MODE_WEAK,
-	FIRE_MODE_STRONG,
-
-	FIRE_MODES_COUNT
-};
```
Comment (mikejsavage) [inline source/gameshared/gs_public.h:663]: extra newline!1111!!!!!!!!!!!!

File: source/gameshared/gs_weapons.cpp [hunk 3]
```diff
@@ -148,7 +148,6 @@ const char *gs_weaponStateNames[] =
 	"WEAPON_STATE_RELOADING",       // clip loading
 	"WEAPON_STATE_NOAMMOCLICK",
 	"WEAPON_STATE_REFIRE",      // projectile loading
-	"WEAPON_STATE_REFIRESTRONG"
```
Comment (mikejsavage) [inline source/gameshared/gs_weapons.cpp:151]: weaponstatenames isn't used anywhere, can delete

File: source/gameshared/gs_public.h (hunk 2 above)
Comment (ObaniGemini) [reply source/gameshared/gs_public.h]: Done

File: source/game/g_utils.cpp (hunk 1 above)
Comment (ObaniGemini) [reply source/game/g_utils.cpp]: Done

**Pull request merged**